robocop/checkers/naming.py

```diff
@@ -45,43 +46,35 @@ def visit_File(self, node):
             suite_name = Path(source).stem
             if '__init__' in suite_name:
                 suite_name = Path(source).parent.name
-            self.check_if_char_in_name(node, suite_name, 'SUITE')
+            for char in suite_name:
+                if char in self.invalid_chars:
+                    self.report("invalid-char-in-name", char, 'suite', node=node)
         super().visit_File(node)
 
     def check_if_char_in_node_name(self, node, name_of_node, is_keyword=False):
-        if is_keyword:
-            matches = [(m.span()) for m in self.var_pattern.finditer(node.name)]
-        else:
-            matches = []
+        variables = find_robot_vars(node.name) if is_keyword else []
         index = 0
-        while index <= len(node.name):
-            index += self._skip_captured_group(index, matches)  # allows to skip ${vars} without breaking indexes
-            if index >= len(node.name):
-                return
-            if node.name[index] in self.invalid_chars:
-                self.report("invalid-char-in-name", node.name[index], self.node_names_map[name_of_node],
-                            node=node,
-                            col=node.col_offset + index + 1)
-            index += 1
-
-    @staticmethod
-    def _skip_captured_group(index, matches):
-        for start, stop in matches:
-            if not start < index >= stop:
-                return stop - start
-        return 0
-
-    def check_if_char_in_name(self, node, name, node_type):
-        for char in self.invalid_chars:
-            if char in name:
-                self.report("invalid-char-in-name", char, self.node_names_map[node_type],
-                            node=node)
+        while index < len(node.name):
+            # skip variables
+            i = 0
+            while variables and i < len(variables):
+                start, stop = variables[i]
+                if start <= index < stop:
+                    index += stop - start
+                    variables.pop(i)
+                    break
+                i += 1
+            else:
```

Comment:
Since variables (start, stop) list will be always ascending I'm also thinking about this approach:
```
if variables:
    start, stop = variables[0]
              if start <= index < stop:
                    index += stop - start
                    variables.pop(0)
     else:

```